app/src/main/java/org/oppia/android/app/hintsandsolution/HintsAndSolutionDialogFragmentPresenter.kt

```diff
@@ -121,26 +115,262 @@ class HintsAndSolutionDialogFragmentPresenter @Inject constructor(
+          if (currentExpandedHintListIndex != null) {
+            bindingAdapter.notifyItemChanged(currentExpandedHintListIndex!!)
+          }
+        }
+      }
+    }
+  }
+
+  private fun bindSolutionViewModel(
+    binding: SolutionSummaryBinding,
+    solutionViewModel: SolutionViewModel
+  ) {
+
+    binding.viewModel = solutionViewModel
+
+    val position: Int = itemList.indexOf(solutionViewModel)
+
+    var isHintListVisible = false
+    if (currentExpandedHintListIndex != null) {
+      isHintListVisible = currentExpandedHintListIndex!! == position
+    }
+    binding.isListExpanded = isHintListVisible
+
+    solutionIndex?.let {
+      isSolutionRevealed?.let {
+        if (solutionIndex == position && isSolutionRevealed!!) {
+          solutionViewModel.isSolutionRevealed.set(true)
+        }
+      }
+    }
+
+    binding.solutionTitle.text = solutionViewModel.title.get()!!.capitalize()
+    // TODO(#1050): Update to display answers for any answer type.
+    if (solutionViewModel.correctAnswer.get().isNullOrEmpty()) {
+      binding.solutionCorrectAnswer.text =
+        """${solutionViewModel.numerator.get()}/${solutionViewModel.denominator.get()}"""
+    } else {
+      binding.solutionCorrectAnswer.text = solutionViewModel.correctAnswer.get()
+    }
+    binding.solutionSummary.text = htmlParserFactory.create(
+      resourceBucketName, entityType, viewModel.explorationId.get()!!, /* imageCenterAlign= */ true
+    ).parseOppiaHtml(
+      solutionViewModel.solutionSummary.get()!!, binding.solutionSummary
+    )
+
+    if (solutionViewModel.solutionCanBeRevealed.get()!!) {
+      binding.root.visibility = View.VISIBLE
+      binding.revealSolutionButton.setOnClickListener {
+        showRevealSolutionDialogFragment()
+      }
+    }
+
+    binding.root.setOnClickListener {
+      if (solutionViewModel.isSolutionRevealed.get()!!) {
+        val previousIndex: Int? = currentExpandedHintListIndex
+        currentExpandedHintListIndex =
+          if (currentExpandedHintListIndex != null && currentExpandedHintListIndex == position) {
+            null
+          } else {
+            position
+          }
+        expandedHintListIndexListener.onExpandListIconClicked(currentExpandedHintListIndex)
+        if (previousIndex != null &&
+          currentExpandedHintListIndex != null &&
+          previousIndex == currentExpandedHintListIndex
+        ) {
+          bindingAdapter.notifyItemChanged(currentExpandedHintListIndex!!)
+        } else {
+          if (previousIndex != null) {
+            bindingAdapter.notifyItemChanged(previousIndex)
+          }
+          if (currentExpandedHintListIndex != null) {
+            bindingAdapter.notifyItemChanged(currentExpandedHintListIndex!!)
+          }
+        }
+      }
+    }
+  }
+
   private fun handleAllHintsExhausted(allHintsExhausted: Boolean) {
-    hintsAndSolutionAdapter.setSolutionCanBeRevealed(allHintsExhausted)
+    if (itemList[itemList.size - 2] is SolutionViewModel) {
```

Comment:
Adding @rt4914 for better naming of const.

Reply:
For here and other places wherever we have used 2 do this:
///
/** RecyclerView items are 2 times of (No. of Hints + Solution), this is because in UI after each hint or solution there is a horizontal line/view which is considered as a separate item in recyclerview. */
`const val RECYCLERVIEW_INDEX_CORRECTION_MULTIPLIER = 2`
///

Now at all places instead of using `2`, write it as `1 * RECYCLERVIEW_INDEX_CORRECTION_MULTIPLIER`


WDYT?

Reply:
sounds good, done

Reply:
Please shift this to `HintsViewModel` 
Ref - https://github.com/oppia/oppia-android/pull/3213#issuecomment-845833128

Reply:
Done

Reply:
resolving on behalf of Farees